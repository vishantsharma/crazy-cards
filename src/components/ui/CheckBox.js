import React from "react";

const Checkbox = ({ name, label, checked, onClick, ...rest }) => {
  return (
    <div className="custom-control custom-checkbox">
      <input
        type="checkbox"
        className="custom-control-input"
        id={name}
        name={name}
        checked={checked}
        onChange={onClick}
        {...rest}
      />
      <label className="custom-control-label" htmlFor={name}>
        {label}
      </label>
    </div>
  );
};

export default Checkbox;
