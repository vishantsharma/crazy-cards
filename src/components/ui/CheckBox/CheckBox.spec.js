import React from 'react';
import TestRenderer from 'react-test-renderer';
import CheckBox from './CheckBox';

describe('CheckBox', () => {
    let props = {
        name: '', 
        label: '', 
        checked: '', 
        onClick: jest.fn().mockReturnValue()
    }
  it('should render', () => {
    const component = TestRenderer.create(<CheckBox {...props} />)
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})