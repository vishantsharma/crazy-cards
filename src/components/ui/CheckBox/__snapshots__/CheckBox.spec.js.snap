// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox should render 1`] = `
Array [
  <input
    checked=""
    id=""
    name=""
    onChange={[MockFunction]}
    type="checkbox"
  />,
  <label
    htmlFor=""
  >
    
  </label>,
]
`;
